### DISCUSSION TITLE
`File` API on Android 11

### DISCUSSION THREAD

**Discussion by fm-sys** [cgeo/cgeo#12964] [General]
Does Google have re-enabled the File API on Android 11? 

At least I read something like that in their docs... 

https://developer.android.com/training/data-storage/shared/media#direct-file-paths

>To help your app work more smoothly with third-party media libraries, Android 11 (API level 30) and higher allow you to use APIs other than the MediaStore API to access media files from shared storage. You can instead access media files directly using either of the following APIs:
>
>   - The File API.
>   - Native libraries, such as fopen().

It would be interesting whether the File API has a better performance than the MediaStore SAF stuff or whether it reacts slow as well...

**Comment by eddiemuc** [comment]
A bit lower the page clarifies that one can only access media files which are attributed to the app (usually the app which created the file). On the linked page it also says that each media file can only be attributed to one app. 

https://developer.android.com/training/data-storage/shared/media#app-attribution

So this approach is not usable to share files between apps. But sharing files is the whole purpose and reason why c:geo implemented SAF. If we don't want to share files we could simply store them in app-private storage (and use File API for it).

**Reply by fm-sys** [reply]
Ah, ok. Thanks for clarification! Although I wonder why they say it is only available starting with Android 11... 

Anyway, I never have seen such a confusing storage management. It's kinda sad that Google couldn't come up with a better solution.

**Reply by eddiemuc** [reply]
> Anyway, I never have seen such a confusing storage management. It's kinda sad that Google couldn't come up with a better solution.

Very true. I never understood why they didn't integrate this "under the hood". (e.g. behind the file API). Also I don't understand why performance is so terrible
